### ISSUE TITLE
stillleben install failed with "RuntimeError: Failed to call ninja -- Could NOT find CUDA"

### ISSUE THREAD

**Issue by ailabktw** [AIS-Bonn/stillleben#3]
Hi, 
thanks to share this wonderful project.

i'm building stillleben in 
- ubuntu 18.04, 5.4.0.52-generic
- GTX 1080
- NVIDIA Driver Version 418.152.00
- CUDA 10.1
- cuDNN 8.0.4.30
- anaconda python 3.7
- pyTorch 1.6.0
- python 3.7.9
and once, it  worked.

However,
i'm suffering the problem with reinstalling stillleben in another pc ( env, spec is same)
"RuntimeError: Failed to call ninja    -- Could NOT find CUDA". 

it can maybe torch issue that i don't know exactly, but nvcc -V & nvidia-smi is work well.
my installed torch said

>\>>> torch.__version__
'1.6.0'
\>>> torch.cuda.is_available()
True
\>>> torch.backends.cudnn.enabled
True

I tried stillleben's installation guide but it failed so i tried some variations.
>$ conda install -c pytorch magma-cuda102 && git clone --recursive --depth 1 -b v1.5.0 https://github.com/pytorch/pytorch.git
& torch's installation guide ( build from sorce & build conda & pip )
& python 3.6, 3.7, 3.8
& cmake 3.18, 3.14
& adding environment variable USE_CUDA=1, CUDA_HOME setting
& python setup.py install & python setup.py clean install & --user option

and this is the error in console.

>(stillleben_build) ailab0@ailab0:~/00.program_ws/stillleben$ python setup.py clean install --user
running clean
running install
running build_deps
-- Found EGL, using that one...
-- Found Corrade: /home/ailab/00.program_ws/stillleben/contrib/corrade/src  found components: Containers rc Utility 
-- Found Corrade: /home/ailab/00.program_ws/stillleben/contrib/corrade/src  found components: Containers rc Utility Main TestSuite 
-- Found Corrade: /home/ailab/00.program_ws/stillleben/contrib/corrade/src  found components: Containers rc Utility PluginManager 
-- Found Corrade: /home/ailab/00.program_ws/stillleben/contrib/corrade/src  found components: Containers rc Utility 
-- Found Magnum: /home/ailab/00.program_ws/stillleben/contrib/magnum/src   
-- Found Corrade: /home/ailab/00.program_ws/stillleben/contrib/corrade/src  found components: Containers rc Utility PluginManager 
-- Found Magnum: /home/ailab/00.program_ws/stillleben/contrib/magnum/src  found components: Trade AnyImageImporter 
-- Found Magnum: /home/ailab/00.program_ws/stillleben/contrib/magnum/src  found components: Trade 
-- Found Magnum: /home/ailab/00.program_ws/stillleben/contrib/magnum/src  found components: Trade AnyImageImporter 
-- Found Corrade: /home/ailab/00.program_ws/stillleben/contrib/corrade/src  found components: Containers rc Utility 
-- Found Magnum: /home/ailab/00.program_ws/stillleben/contrib/magnum/src   
-- Found Magnum: /home/ailab/00.program_ws/stillleben/contrib/magnum/src  found components: GL 
-- Found Corrade: /home/ailab/00.program_ws/stillleben/contrib/corrade/src  found components: Containers rc Utility PluginManager 
-- Found Magnum: /home/ailab/00.program_ws/stillleben/contrib/magnum/src  found components: Trade GL MeshTools Primitives SceneGraph Shaders DebugTools WindowlessEglApplication missing components: Shapes
-- Could NOT find CUDA (missing: CUDA_INCLUDE_DIRS) (found version "10.1")
-- Found PyTorch 1.6.0.
--  - CUDA support: ON
--  - c10 lib: /home/ailab/anaconda3/envs/stillleben_build/lib/python3.7/site-packages/torch/lib/libc10.so
--  - torch lib: /home/ailab/anaconda3/envs/stillleben_build/lib/python3.7/site-packages/torch/lib/libtorch.so
--  - torch_python lib: /home/ailab/anaconda3/envs/stillleben_build/lib/python3.7/site-packages/torch/lib/libtorch_python.so
-- Building Differentiation Modules
CMake Error at /home/ailab/anaconda3/envs/stillleben_build/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find CUDA (missing: CUDA_INCLUDE_DIRS) (found version "10.1")
Call Stack (most recent call first):
  /home/ailab/anaconda3/envs/stillleben_build/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  /home/ailab/anaconda3/envs/stillleben_build/share/cmake-3.18/Modules/FindCUDA.cmake:1119 (find_package_handle_standard_args)
  python/CMakeLists.txt:96 (find_package)

>-- Configuring incomplete, errors occurred!
See also "/home/ailab/00.program_ws/stillleben/python/cpp_build/CMakeFiles/CMakeOutput.log".
Failed to run "cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX=/home/ailab/00.program_ws/stillleben/python/stillleben -DUSE_RELATIVE_RPATH=ON -GNinja ../.. -DEXTRA_RPATH=/home/ailab/anaconda3/envs/stillleben_build/lib"
ninja: error: loading 'build.ninja': No such file or directory
Traceback (most recent call last):
  File "setup.py", line 115, in <module>
    'dist_dir': 'python/dist'
  File "/home/ailab/.local/lib/python3.7/site-packages/setuptools/__init__.py", line 153, in setup
    return distutils.core.setup(**attrs)
  File "/home/ailab/anaconda3/envs/stillleben_build/lib/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/home/ailab/anaconda3/envs/stillleben_build/lib/python3.7/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/home/ailab/anaconda3/envs/stillleben_build/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "setup.py", line 82, in run
    self.run_command('build_deps')
  File "/home/ailab/anaconda3/envs/stillleben_build/lib/python3.7/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/home/ailab/anaconda3/envs/stillleben_build/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "setup.py", line 77, in run
    build_stillleben()
  File "setup.py", line 52, in build_stillleben
    raise RuntimeError('Failed to call ninja')
RuntimeError: Failed to call ninja

second thing,
other env is once installed. (i don't know how installed in this env ... i really don't know what is difference ) 

but when i launch python script, it got error like this

>(stillleben) ailab@ailab:~/00.program_ws/stillleben/examples$ python viewer.py 
Traceback (most recent call last):
  File "viewer.py", line 6, in <module>
    import stillleben as sl
  File "/home/ailab/00.program_ws/stillleben/python/stillleben/__init__.py", line 9, in <module>
    from . import diff
  File "/home/ailab/00.program_ws/stillleben/python/stillleben/diff.py", line 23, in <module>
    from .lib.libstillleben_python import Scene, RenderPassResult
ModuleNotFoundError: No module named 'stillleben.lib'

thank for reading this short lack of my language (...) and
I'd appreciate it if you could make a advice.

**Comment by xqms** [comment]
Hey, it seems your CUDA installation is incomplete. Probably you have the CUDA runtime installed, but not the development headers. That's why it's complaining about `CUDA_INCLUDE_DIRS`.

Can you post the output of `ls -l /usr/local` and `dpkg -l | grep cuda`?

**Comment by ailabktw** [comment]
Thanks for reply!


here is result of `ls -l /usr/local`

>total 36
drwxr-xr-x  2 root root 4096  2 10  2019 bin
lrwxrwxrwx  1 root root   21 10 22 13:18 cuda -> /usr/local/cuda-10.1/
drwxr-xr-x 17 root root 4096 10 22 13:18 cuda-10.1
drwxr-xr-x  2 root root 4096  2 10  2019 etc
drwxr-xr-x  2 root root 4096  2 10  2019 games
drwxr-xr-x  3 root root 4096  9 16 19:55 include
drwxr-xr-x  5 root root 4096  9 16 19:55 lib
lrwxrwxrwx  1 root root    9  8 27 11:44 man -> share/man
drwxr-xr-x  2 root root 4096  2 10  2019 sbin
drwxr-xr-x  8 root root 4096  8 27 19:29 share
drwxr-xr-x  2 root root 4096  2 10  2019 src

and
the result of `dpkg -l | grep cuda` is nothing else.
i just installed it from .run file and followed doc

am i followed something wrong...?
thanks you

**Comment by xqms** [comment]
Hmm, everything looks ok to me, though I never installed CUDA using the `.run` installer. I always use the debian packages. Which `.run` file did you install exactly?

**Comment by ailabktw** [comment]
for now, I installed ` cuda_10.1.243_418.87.00_linux.run` 

but, I tried `cuda_10.1.105_418.39_linux.run` and the result is same.
when I install the `runfile`, I tried skipping the `418 driver` (using `ubuntu default 418` & using `450, 440`) or not (installing 418 driver with CUDA) and the result is same too.

It's very strange... every program(cuda, tensorflow, torch .. etc what i use) is function well but I can't install only stillleben... (in this time.. it once functioned well)

I'll try installing ubuntu one more or installing CUDA with `.deb`. and tell you the result later.
it maybe something twisted when i reinstalling system...
Thank you for your guidance. :)
